### ISSUE TITLE
Multiplex Model Calls In Worker Threads

### ISSUE THREAD

**Issue by definitelysean** [compbiomed/celda#127]
Currently, calling the celda() function spins off threads for all of the requested model (e.g. celda_CG) objects. The downside to this is that the celda_C(), celda_G(), etc. functions have to have a lot of common boilerplate, which isn't very DRY.

A nicer way to do this would be to have threads start with a function to set up the "environment" (e.g. logging), which subsequently calls the desired model.

Basically, we should implement a layer of abstraction between calling celda() and running the actual models, and this layer should be the top of each doParallel thread's execution.
